src/cli/cli.cpp

```diff
@@ -1630,6 +1630,46 @@ void Interpreter::HandleDnsServiceResponse(otError aError, const otDnsServiceRes
 #endif // OPENTHREAD_CONFIG_DNS_CLIENT_ENABLE
 
 #if OPENTHREAD_FTD
+const char *EidCacheStateToString(otCacheEntryState aState)
+{
+    static const char *const kStateStrings[4] = {
+        "cache",
+        "snoop",
+        "query",
+        "retry",
+    };
+
+    return aState <= OT_CACHE_ENTRY_STATE_RETRY_QUERY ? kStateStrings[aState] : "unkno";
```

Comment:
```suggestion
    return aState < OT_ARRAY_LENGTH(kStateStrings) ? kStateStrings[aState] : "unkno";
```

in case we change the state values.

Reply:
I had to add `static_cast` to avoid the following:
```
../../src/cli/cli.cpp:1642:19: error: result of comparison of constant 4 with expression of type 'otCacheEntryState' is always true [-Werror,-Wtautological-constant-out-of-range-compare]
    return aState < OT_ARRAY_LENGTH(kStateStrings) ? kStateStrings[aState] : "unknown";
           ~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```